src/Core/System/Consent/Api/ConsentController.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Consent\Api;
+
+use Shopware\Core\Framework\Api\ApiException;
+use Shopware\Core\Framework\Api\Context\AdminApiSource;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Routing\ApiRouteScope;
+use Shopware\Core\PlatformRequest;
+use Shopware\Core\System\Consent\ConsentContext;
+use Shopware\Core\System\Consent\ConsentScope;
+use Shopware\Core\System\Consent\Service\ConsentService;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @internal
+ */
+#[Package('data-services')]
+#[Route(defaults: [PlatformRequest::ATTRIBUTE_ROUTE_SCOPE => [ApiRouteScope::ID]])]
+class ConsentController
+{
+    public function __construct(
+        private readonly ConsentService $consentService,
+    ) {
+    }
+
+    #[Route(path: '/api/consents', name: 'api.consents.fetch', defaults: ['auth_required' => true], methods: ['GET'])]
```

Comment:
It seems like we don't rely on the DAL for storing the consent info, that means we need to take of ACL manually, so we need add custom ACL permissions for those APIs